lib/resty/upload.lua

```diff
@@ -21,12 +24,52 @@ local STATE_READING_HEADER = 2
 local STATE_READING_BODY = 3
 local STATE_EOF = 4
 
+local function warped_receiveuntil(self, until_str)
+    local iter, err_outer = self:old_receiveuntil(until_str)
+    if iter == nil then
+        ngx_finish_body()
+    end
+
+    local function warped(size)
+        local ret, err = iter(size)
+        if ret then
+            ngx_append_body(ret)
+        end
+        -- non-nil ret for call with no size or successful size call and nil ret
```

Comment:
you can put one extra blank line here before to make the code easier to read
```suggestion

        -- non-nil ret for call with no size or successful size call and nil ret
```